### PR TITLE
Fixes #36899 - Add Dependabot Configuration

### PR DESCRIPTION
**Pull request by archanaserver**
This PR adds the Dependabot configuration for automated dependency management. Dependabot will help us keep smart-proxy bundler dependencies up to date.

The configuration is located in `.dependabot/config.yml`.

### CODE REVIEW

**Comment by archanaserver** [comment]
@ekohl Do you want me to add any specific config here?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  - package-ecosystem: 'bundler'
+    directory: '/'
+    schedule:
+      interval: 'daily'
+    open-pull-requests-limit: 10
```
Comment (evgeni) [inline .github/dependabot.yml]: is that necessary? we don't set it in the other repos and so far didn't have any PR floods.

**Review by evgeni — APPROVED** [review]
I wonder if this will work with our
https://github.com/theforeman/smart-proxy/blob/54d4f0f10866db9c1fa630c79c774223f1b55119/Gemfile#L7-L9

But Bundler has an [`eval_gemfile` function since 1.3](https://github.com/rubygems/bundler/commit/c6100f6bd246a8259f159285967f075b592e80c9) which we could use instead, and that is (according to tests in https://github.com/dependabot/dependabot-core/tree/main/bundler/spec/fixtures/projects/bundler2) supported by dependabot.

Let's try this!

File: .github/dependabot.yml (hunk 1 above)
Comment (ekohl) [reply .github/dependabot.yml]: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#open-pull-requests-limit says the default limit is 5. I don't think we need to increase that to 10 so :+1: for removing this part.

File: Gemfile [hunk 2]
```diff
@@ -5,5 +5,5 @@ gemspec
 gem 'concurrent-ruby', '~> 1.0', require: 'concurrent'
 
 Dir[File.join(__dir__, 'bundler.d', '*.rb')].each do |bundle|
-  instance_eval(Bundler.read_file(bundle))
+  Bundler.eval_gemfile(bundle)
```
Comment (ekohl) [inline Gemfile]:
```
[!] There was an error parsing `Gemfile`: undefined method `eval_gemfile' for Bundler:Module. Bundler cannot continue.
```

You should be able to reproduce this locally with `bundle install`. This works for me
```suggestion
  eval_gemfile(bundle)
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 2 above)
Comment (archanaserver) [reply Gemfile]: Yes, Thank you.

**Review by ekohl — APPROVED** [review]
Let's try this.

**Pull request merged**